FlashCards.HopelessCoding/FlashCards.HopelessCoding/DTOs/FlashcardService.cs

```diff
@@ -0,0 +1,104 @@
+﻿using Spectre.Console;
+using System.Data.SqlClient;
+
+namespace FlashCards.HopelessCoding.DTOs;
+
+public class FlashcardService
+{
+    private readonly string _connectionString;
+    private static Dictionary<int, int> _displayIdToActualId = new Dictionary<int, int>();
+
+    public FlashcardService(string connectionString)
+    {
+        _connectionString = connectionString;
+    }
+
+    public Dictionary<int, int> GetDisplayIdToActualId()
+    {
+        return _displayIdToActualId;
+    }
+
+    public List<FlashcardDto> GetFlashcards(string stackName, int? amount)
+    {
+        List<FlashcardDto> flashcards = new List<FlashcardDto>();
+
+        string query = "SELECT " + (amount != null ? $"TOP {amount}" : "") + " * FROM FlashCards " +

```

Comment:
You're doing well to prevent [SQL Injection](https://www.w3schools.com/sql/sql_injection.asp) when you insert the stack name in the query. Don't forget to do it for all parameters!